src/modules/hip/kernel/color_temperature.hpp

```diff
@@ -0,0 +1,236 @@
+    color_temperature_hip_compute(srcPtr, &pix_f24, &adjustmentValue_f4);
+    rpp_hip_pack_float24_pln3_and_store24_pln3(dstPtr + dstIdx, dstStridesNCH.y, &pix_f24);
+}
+
+template <typename T>
+__global__ void color_temperature_pln3_pkd3_tensor(T *srcPtr,
+                                                   uint3 srcStridesNCH,
+                                                   T *dstPtr,
+                                                   uint2 dstStridesNH,
+                                                   int *adjustmentValueTensor,
+                                                   RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+    {
+        return;
+    }
+
+    uint srcIdx = (id_z * srcStridesNCH.x) + ((id_y + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNCH.z) + (id_x + roiTensorPtrSrc[id_z].xywhROI.xy.x);
+    uint dstIdx = (id_z * dstStridesNH.x) + (id_y * dstStridesNH.y) + id_x * 3;
+
+    float4 adjustmentValue_f4 = (float4)((float)adjustmentValueTensor[id_z]);
+
+    d_float24 pix_f24;
+
+    rpp_hip_load24_pln3_and_unpack_to_float24_pln3(srcPtr + srcIdx, srcStridesNCH.y, &pix_f24);
+    color_temperature_hip_compute(srcPtr, &pix_f24, &adjustmentValue_f4);
+    rpp_hip_pack_float24_pln3_and_store24_pkd3(dstPtr + dstIdx, &pix_f24);
+}
+
+template <typename T>
+RppStatus hip_exec_color_temperature_tensor(T *srcPtr,
+                                            RpptDescPtr srcDescPtr,
+                                            T *dstPtr,
+                                            RpptDescPtr dstDescPtr,
+                                            RpptROIPtr roiTensorPtrSrc,
+                                            RpptRoiType roiType,
+                                            rpp::Handle& handle)
+{
+    if (roiType == RpptRoiType::LTRB)
+        hip_exec_roi_converison_ltrb_to_xywh(roiTensorPtrSrc, handle);
+
+    if ((srcDescPtr->c == 3) && (dstDescPtr->c == 3))
+    {
+        int localThreads_x = LOCAL_THREADS_X;
+        int localThreads_y = LOCAL_THREADS_Y;
+        int localThreads_z = LOCAL_THREADS_Z;
+        int globalThreads_x = (dstDescPtr->strides.hStride + 7) >> 3;
+        int globalThreads_y = dstDescPtr->h;
+        int globalThreads_z = handle.GetBatchSize();
+
+        if ((srcDescPtr->layout == RpptLayout::NHWC) && (dstDescPtr->layout == RpptLayout::NHWC))
+        {
+            globalThreads_x = (dstDescPtr->strides.hStride / 3 + 7) >> 3;
+            hipLaunchKernelGGL(color_temperature_pkd_tensor,
+                               dim3(ceil((float)globalThreads_x/localThreads_x), ceil((float)globalThreads_y/localThreads_y), ceil((float)globalThreads_z/localThreads_z)),
```

Comment:
Use dim3(ceil((float)globalThreads_x/LOCAL_THREADS_X... directly here, and remove L168,169,170

Reply:
Done